### PR TITLE
Fixed #55 , send fail due to api change.

### LINKED ISSUE #55: new facebook api - FIXED !!!
Hey guys, I use fbchat for a couple of weeks now and everything worked fine.

 But since 23h I can't send any msg :/

import fbchat
client = fbchat.Client("....", ".....")
sent = client.send(100001986531748, "test123")
if sent:
    print("Message sent successfully!")
print("nope! test")

In cmd:

D:\Spiele\TEST>python test.py
Logging in...
nope! test

When loggin in the fbacc, I can msg everyone without a problem. Furthermore I can get the last messages sent without a problem

Any ideas ? Sorry I am new to python

thx in advance <3

### LINKED ISSUE #57: [FIXED] fbid in _parseMessage is always the other user's Id
the actual author's id is in `metadata['delta']['messageMetadata']['actorFbId']` . Sending PR

### PR DESCRIPTION
**Pull request by thekindlyone**
**DIGEST OF CHANGES**

**clients.py**:

Cloned data to be exact payload the js project is sending in send to fix #55 

Changed fbid in _parseMessage to fix #57 

If debug then print payload that is being sent on send

Added getUserInfo to add feature request #47 

**utils.py** :

Added generateOfflineThreadingID and getSignatureID which are clones of the same code in js project

**readme**  :
Added Echobot example 

Added getUserInfo example

### CODE REVIEW

**Comment by stefanhdao** [comment]
Just figured it out

It's the 'offline_threading_id' and 'threading_id' that needed to be added.

File: fbchat/client.py [hunk 1]
```diff
@@ -221,37 +222,74 @@ def send(self, recipient_id, message=None, message_type='user', like=None, image
         else:
             thread_id = None
             user_id = recipient_id
-
+        # messageAndOTID=generateMessageID(self.client_id)
```
Comment (Schmavery) [inline fbchat/client.py]: Why are all these lines commented rather than deleted?

**Comment by thekindlyone** [comment]
ah the I forgot to delete those lines. commented when testing. This hasn't been merged.
do I send another pull request with that cleaned up or what?

**Comment by JohnathonNow** [comment]
Do you want to also remove the `message_batch[0]` bit from the image_id and sticker_id lines?

**Comment by Schmavery** [comment]
Might be worth creating separate PRs for solutions to different issues in the future :)
Not sure how the owner of this repo likes to handle it but it can make it easier to merge things.

**Comment by thekindlyone** [comment]
Damnit, I even asked another friend to confirm protocol on this. He advised me to just push to master. Henceforth, shall keep in mind.

**Comment by thekindlyone** [comment]
@JohnathonNow , I'll have to test the image feature first. Will do, tomorrow.

**Comment by JohnathonNow** [comment]
Here's a simple program for just sending an image by URL:
```
import fbchat

USER = ...
PASS = ...
FRIEND = ...
MESSAGE = 'Look at this cute picture I found!'
IMAGE_URL = 'https://upload.wikimedia.org/wikipedia/commons/thumb/2/2e/Pi-symbol.svg/2000px-Pi-symbol.svg.png'

client = fbchat.Client(USER, PASS)
CHAT_ID = client.getUsers(FRIEND)[0].uid
client.sendRemoteImage(CHAT_ID, MESSAGE, 'user', IMAGE_URL)
```

**Pull request merged**